### ISSUE TITLE
Find compiler classes

### ISSUE THREAD

**Issue by thekid** [xp-framework/unittest#26]
```bash
$ find src/test/php/
src/test/php/
src/test/php/de
src/test/php/de/thekid
src/test/php/de/thekid/dialog
src/test/php/de/thekid/dialog/ImportTest.php

$ xp -m ../xp/compiler/ test src/test/php/
*** Error: Cannot find any test cases in FileSystemCL<./src/test/php>
```

Instead of crawling the filesystem itself unittest should use the class loading mechanism!

**Comment by thekid** [comment]
Problem is the class loading mechanism doesn't provide a way to map paths to class files except for the non-generic solution seen in `FolderSource` which checks for `FileSystemClassLoaders`'s *path* property. 

We can't just use `packageContents()` because that might pick up classes outside of the given folder!

**Comment by thekid** [comment]
Here's an non-generic solution (on top of the already non-generic solution):

```diff
$ git diff
diff --git a/src/main/php/xp/unittest/sources/FolderSource.class.php b/src/main/php/xp/unittest/sources/FolderSource.class.php
index 7d939b1..47c664b 100755
--- a/src/main/php/xp/unittest/sources/FolderSource.class.php
+++ b/src/main/php/xp/unittest/sources/FolderSource.class.php
@@ -44,7 +44,9 @@ class FolderSource extends AbstractSource {
           yield $entry;
         }
       } else if (0 === substr_compare($entry->name(), \xp::CLASS_FILE_EXT, $e)) {
-        yield $entry;
+        yield substr($entry, 0, $e);
+      } else if (0 === substr_compare($entry->name(), '.php', -4)) {
+        yield substr($entry, 0, -4);
       }
     }
   }
@@ -60,10 +62,10 @@ class FolderSource extends AbstractSource {
   public function provideTo($suite, $arguments) {
     $empty= true;

+    $cl= ClassLoader::getDefault();
     $l= strlen($this->loader->path);
-    $e= -strlen(\xp::CLASS_FILE_EXT);
     foreach ($this->classFilesIn(new Folder($this->loader->path)) as $classFile) {
-      $class= $this->loader->loadClass(strtr(substr($classFile, $l, $e), DIRECTORY_SEPARATOR, '.'));
+      $class= $cl->loadClass(strtr(substr($classFile, $l), DIRECTORY_SEPARATOR, '.'));
       if ($class->isSubclassOf(TestCase::class) && !Modifiers::isAbstract($class->getModifiers())) {
         $suite->addTestClass($class, $arguments);
         $empty= false;
```

**Comment by thekid** [comment]
Implemented in xp-framework/unittest@98d6bb4

**Comment by thekid** [comment]
:shipit: Shipped in https://github.com/xp-framework/unittest/releases/tag/v9.4.0
